elastic/beats
# Added PFlog support

Enhancement

## What does this PR do?

Adds support for sniffing PFLog packets in Packetbeat

## Why is it important?

PFlog is a packet filter (firewall) available in some operating systems like FreeBSD, OpenBSD and macOS. PFLog interface can be used to log packets directly as opposed to [pflogd](https://man.openbsd.org/pflogd). 

## Checklist



- [x] I have added an entry in `CHANGELOG.next.asciidoc`


## How to test this PR locally

An easiest example configuration of pf.conf would be 
```
pass log all
```
Then create `pflog0` interface using `ifconfig` or `ip` commands
After that you can enable pf by running 
```
sudo pfctl -f /etc/pf.conf
```
You should now be able to sniff packets using tcpdump:
```
tcpdump -n -e -ttt -i pflog0
```
## Related issues

Discussed in pull #1362

## Other changes
Migrated `github.com/tsg/gopacket` to `github.com/google/gopacket`
